smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/PublisherDecorator.java

```diff
@@ -0,0 +1,20 @@
+package io.smallrye.reactive.messaging;
+
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.eclipse.microprofile.reactive.streams.operators.PublisherBuilder;
+
+/**
+ * SPI to allow extension of publishers included in the final graph
+ */
+public interface PublisherDecorator {
+
+    /**
+     * Decorate a publisher
+     * 
+     * @param publisher the publisher to decorate
+     * @param the name of the channel to which this publisher publishes
+     * @return the extended publisher
+     */
+    public PublisherBuilder<? extends Message> decoratePublisher(PublisherBuilder<? extends Message> publisher, String name);
```

Comment:
`decorate` is shorter and good enough.
Actually the current name is misleading because it gets a PublisherBuilder.

Reply:
+1

Reply:
Done